public/app/plugins/panel/graph/graph.ts

```diff
@@ -428,7 +452,9 @@ coreModule.directive('grafanaGraph', function($rootScope, timeSrv, popoverSrv) {
         let defaultTicks = panelWidth / 50;
 
         if (data.length && bucketSize) {
-          ticks = _.map(data[0].data, point => point[0]);
+          ticks = _.chain(data).map(d => {
+            return _.map(d.data, point => point[0]);
```

Comment:
actually even in your test the for loop is 12 times faster (6,749 ops per second vs 557 ops per second with the map_uniq version)